rapids-cmake/cpm/detail/gpv_write_file.cmake

```diff
@@ -0,0 +1,252 @@
+#=============================================================================
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+include_guard(GLOBAL)
+
+#[=======================================================================[.rst:
+rapids_cpm_gpv_extract_source_git_info
+-----------------------------------------
+
+.. versionadded:: v24.04.00
+
+
+#]=======================================================================]
+function(rapids_cpm_gpv_extract_source_git_info package git_url_var git_sha_var)
+  set(source_dir "${CPM_PACKAGE_${package}_SOURCE_DIR}")
+  set(_RAPIDS_URL)
+  set(_RAPIDS_SHA)
+  if(EXISTS "${source_dir}")
+    execute_process(COMMAND ${GIT_EXECUTABLE} ls-remote --get-url
+                    WORKING_DIRECTORY ${source_dir}
+                    ERROR_QUIET
+                    OUTPUT_VARIABLE _RAPIDS_URL
+                    OUTPUT_STRIP_TRAILING_WHITESPACE)
+    # Need to handle when we have applied N patch sets to the git repo and therefore the latest
+    # commit is just local
+    #
+    # Find all commits on our branch back to the common parent ( what we cloned )
+    #
+    execute_process(COMMAND ${GIT_EXECUTABLE} show-branch --current --sha1-name
```

Comment:
Can we use `git rev-parse HEAD` to get the SHA? I don't understand the need for branch information here.

Reply:
As mentioned in the above comment. We can't use `git rev-parse HEAD` since rapids-cmake supports applying `git format-patch` patches which cause local only commits to exist.